pyccel/ast/operators.py

```diff
@@ -784,6 +784,8 @@ def __new__(cls, arg1 = None, arg2 = None, simplify = False):
                 dtype = cls._calculate_type(arg1, arg2)
                 return convert_to_literal(arg1.python_value - arg2.python_value,
                                           dtype)
+            if isinstance(arg1, PyccelAdd) and arg1.args[1] == arg2:
+                return arg1.args[0]
```

Comment:
How about the similar case where the first argument of `PyccelAdd` is equal to `arg2`?

```suggestion
            if isinstance(arg1, PyccelAdd):
                if arg1.args[1] == arg2:
                    return arg1.args[0]
                elif arg1.args[0] == arg2:
                    return arg1.args[1]
```